system/Database/BasePreparedQuery.php

```diff
@@ -147,17 +147,26 @@ abstract public function _execute(array $data): bool;
     abstract public function _getResult();
 
     /**
-     * Explicitly closes the statement.
+     * Explicitly closes the prepared statement.
      */
-    public function close()
+    public function close(): bool
     {
-        if (! is_object($this->statement) || ! method_exists($this->statement, 'close')) {
-            return;
+        if (! isset($this->statement)) {
+            return true;
```

Comment:
Why return `true`?
`BadMethodCallException`?

Reply:
Done, a `BadMethodCallException` is now thrown when trying to close a non-existing prepared statement

Reply:
Please add `@throws` in the docblock of `close()`

Reply:
Done, docblock added